### PR TITLE
make deployment process concurrent

### PR DESCRIPTION
**Pull request by mariobassem**
## Related issues:

- #174

### CODE REVIEW

File: grid-client/deployer/deployer.go [hunk 1]
```diff
@@ -155,35 +156,37 @@ func (d *Deployer) deploy(
...
 			log.Debug().Uint32("Number of public ips", publicIPCount)
 
-			contractID, err := d.substrateConn.CreateNodeContract(d.identity, node, dl.Metadata, hashHex, publicIPCount, newDeploymentSolutionProvider[node])
-			log.Debug().Uint64("CreateNodeContract returned id", contractID)
-			if err != nil {
-				return currentDeployments, errors.Wrap(err, "failed to create contract")
-			}
+			nodeID := node
+			dep := dl
+			errGroup.Go(func() error {
```
Comment (xmonader) [inline grid-client/deployer/deployer.go:195]: I don't think this would work, as you need to maintain the sequence number update or use batch calls?

Comment (mariobassem) [reply grid-client/deployer/deployer.go:195]: i tested this multiple times manually, and it was working fine...
but using batch calls seems more fun :D
but i think would require some, not so minor, changes in the current deployment process

**Comment by mariobassem** [comment]
deletions and creations work fine, updates operation still needs some fixes.
also, tests don't pass, should have a look into it.

**Comment by xmonader** [comment]
blocking comment for @rawdaGastan and @AbdelrahmanElawady

**Comment by rawdaGastan** [comment]
We now support batch calls for deployments. So, we won't support concurrency for them. The user can use batch calls.
- https://github.com/threefoldtech/tfgrid-sdk-go/issues/116

**Pull request closed without merging**